src/lyra.ts

```diff
@@ -18,6 +20,20 @@ export type PropertiesSchema = {
   [key: string]: PropertyType | PropertiesSchema;
 };
 
+export type TokenizerConfig = {
+  enableStemming?: boolean;
+  enableStopWords?: boolean;
+  customStopWords?: ((stopWords: string[]) => string[]) | string[];
+  stemmingFn?: (word: string) => string;
+};
+
+export type TokenizerConfigExec = {
```

Comment:
You can streamline this as `export type TokenizerConfigExec = Required<TokenizerConfig>`

Reply:
Nice catch!

Reply:
Oh, one problem is that `stemmingFn` can be undefined (to fallback to the default stemming function). `Required` is not compatible with `undefined` values